apps/github-api-integration/README.md

```diff
@@ -0,0 +1,172 @@
+  this.options.headers = {
+    'User-Agent': 'HOF Example API Integration'
+  };
+}
+```
+We can call `super` before setting the header to hand off the creation of `this.options` to the parent, base Model.
+
+### Setting the url
+
+The Model's `url` method will be called when any Model API method is called, but before the request to the Github API is invoked.
+In our example we override the parent Model's `url` method to return the `url` argument we passed to both the behaviours, replacing `:username` with the `name` entered by the user.
+
+```js
+url() {
+  // Augment configured url the user entered `name` param
+  // that we set on the model in the behaviour
+  return this.options.url.replace(':username', this.get('name'));
+}
+```
+
+When fetch is called from either behaviour, our Model's `url` method returns the url with which to make the request to the Github API.
+
+## Behaviours
+
+Each of our behaviours perform discreet functions.
+
+### Validate name
+The `validate-name` behaviour needs to check if the name the user has entered is a real Github username. If it is, the behaviour saves the name on the session or renders a user error if it is not.
+
+The behaviour has one method, `process`, which is called during a POST request when the form is submitted.
+
+```js
+process(req, res, next) {
+  // Set the `name` to a new instance of the Github Model
+  const name = req.form.values.name;
+  const model = new Model({name}, config);
+
+  // Make a GET request to the Github API
+  model.fetch()
+    .then(result => {
+      // the user exists, set the name on the session
+      req.sessionModel.set('name', name);
+      // continue with the request
+      super.process(req, res, next);
+    })
+    .catch(err => {
+      if (err.status === 404) {
+        // the user does not exist
+        // return an error to the user
+        return res.status(400).render('400', {
+          name: name,
+          baseUrl: req.baseUrl
+        });
+      }
+      next(err);
+    });
+}
+```
+
+When the request resolves, it renders with either the template set in the appropriate `step` on success or `400.html` if the request is invalid.
+
+### Fetch repos
+The `fetch-repos` behaviour is designed to retrieve the `name` set previously to the session and use it to GET all repositories from the Github API for that user.
+
+It also has only one method, `getValues`, which is called during a GET request when a form is loaded.
+
+```js
+getValues(req, res, next) {
+  // Set the `name` to a new instance of the Github Model
+  const name = req.sessionModel.get('name');
+  const model = new Model({name}, config);
+
+  // Make a GET request to the Github API
+  return model.fetch()
+    .then(result => {
+      // Assign a subset of the API result
+      // to the locals property
+      res.locals.repos = _.map(result, repo => ({
+        name: repo.name,
+        desc: repo.description,
+        url: repo.html_url
+      }));
```

Comment:
Here I would call back with the repos as values. Then use `{{values.repos}}` in the template.